$OpenBSD: patch-src_common_dtpthread_h,v 1.1 2013/04/26 23:10:26 sthen Exp $

XXX disable the default PTHREAD_MUTEX_STRICT_NP behaviour (sorry!)

--- src/common/dtpthread.h.orig	Wed Apr  3 21:13:14 2013
+++ src/common/dtpthread.h	Fri Apr 26 18:50:17 2013
@@ -75,6 +75,14 @@ dt_pthread_mutex_init_with_caller(dt_pthread_mutex_t *
 {
   memset(mutex, 0x0, sizeof(dt_pthread_mutex_t));
   snprintf(mutex->name, 256, "%s:%d (%s)", file, line, function);
+  if (attr == NULL) {
+    pthread_mutexattr_t a;
+    pthread_mutexattr_init(&a);
+    pthread_mutexattr_settype(&a, PTHREAD_MUTEX_NORMAL);
+    const int ret = pthread_mutex_init(&(mutex->mutex), &a);
+    pthread_mutexattr_destroy(&a);
+    return ret;
+  }
   const int ret = pthread_mutex_init(&(mutex->mutex), attr);
   return ret;
 }
